pkg/backend/filestate/backend.go

```diff
@@ -99,7 +104,16 @@ func New(d diag.Sink, u string) (Backend, error) {
 		return nil, err
 	}
 
-	bucket, err := blob.OpenBucket(context.TODO(), u)
+	blobmux := blob.DefaultURLMux()
+
+	if strings.HasPrefix(u, gcsblob.Scheme) {
+		blobmux, err = GoogleCredentialsMux()
```

Comment:
comment htat gcs needs special support as we want to support a few more credential schemes outside of what gcs natively supports.

Reply:
i'd like this to pass back a bool statin that it's ok that we don't get a permalink.  that way we can just warn in that case, but error for all other cases.

Reply:
I've added the comment. I'm not sure what you mean by the "pass back a bool".

Currently the `GoogleCredentialsMux()` will log the warning if there's not enough info for the permalink.

What do you mean by:

> but error for all other cases

The only case that `GoogleCredentialsMux()` will return an error is when the credentials are structurally corrupt or not found at all. There's not permissions checking or the like here.

The warning about not being able to create a statefile permalink can be done because it's easy to check if the credentials file is missing both `GoogleAccessID` and `PrivateKey`. Go cloud documents that these are required for `SignedURL()`. The reason to log the warning here at all is because this happens before pulumi runs the update, so a user can see the warning before the preview incase it matters to them. It'll obviously log that the permalink can't be made later on if the call to `SignedURL()` fails.